internal/bgp/bgp.go

```diff
@@ -525,8 +527,21 @@ type tcpmd5sig struct {
 // proper TCP MD5 options when the password is not empty. Works by manupulating
 // the low level FD's, skipping the net.Conn API as it has not hooks to set
 // the neccessary sockopts for TCP MD5.
-func dialMD5(ctx context.Context, addr, password string) (net.Conn, error) {
-	laddr, err := net.ResolveTCPAddr("tcp", "[::]:0")
+func dialMD5(ctx context.Context, addr string, srcAddr string, password string) (net.Conn, error) {
```

Comment:
nit: two consecutive params of the same type (addr and srcAddr of type string) can be written to only write the type (string) once. Can you change it?

And we should probably ask a linter to check these things in the future.

Reply:
Fixed.

Happy to add a linter for this. I couldn't find a linter which checks for this case though. Do you know one @rata?